### DISCUSSION TITLE
Do I need to call Rows.Close() after using db.ScanRows() ?

### DISCUSSION THREAD

**Discussion by siraphobk** [uptrace/bun#598] [Q&A] [answered]
To quote the documentation https://bun.uptrace.dev/guide/queries.html#scanning-rows.

```go
rows, err := db.QueryContext(ctx, "SELECT * FROM users")
if err != nil {
    panic(err)
}

err = db.ScanRows(ctx, rows, &users)
```

I'm not sure that should I call `defer rows.Close()` after using `db.QueryContext`? Does the close mechanism handled in `ScanRows` automatically?

**Comment by vmihailenco** [comment, marked as answer]
`ScanRows` closes the `rows` for you.

**Reply by siraphobk** [reply]
Thanks a bunch!
